src/stores/datStore.ts

```diff
@@ -407,19 +410,50 @@ export default class DatStore {
         numToGet: number
     ): Promise<TradeEvent[]> {
         const { library } = this.rootStore.providerStore.getActiveWeb3React();
+        let toBlock = await library.eth.getBlockNumber();
         let fromBlock = (this.rootStore.configStore.getDATinfo()).fromBlock;
-        if (fromBlock > TRADES_FROM_LAST_BLOCKS)
-          fromBlock = (await library.eth.getBlockNumber()) - TRADES_FROM_LAST_BLOCKS;
-          
-        const buyEvents = await this.fetchBuyEvents(datAddress, numToGet, fromBlock);
-        const sellEvents = await this.fetchSellEvents(datAddress, numToGet, fromBlock);
-
-        let combinedTrades: any[] = buyEvents.concat(sellEvents);
-        combinedTrades = combinedTrades.sort(function (a, b) {
-            return b.blockNumber - a.blockNumber;
+        let blockLimit = (this.rootStore.configStore.getDATinfo()).fromBlock;
+        let tradesToReturn = [];
+        
+        if (toBlock > BLOCK_PER_TRADES_FETCH) {
+          blockLimit = toBlock - TRADES_FROM_LAST_BLOCKS;     
+        }
+        if (toBlock > BLOCK_PER_TRADES_FETCH) {
```

Comment:
Can these two conditions be combined into one?

Reply:
they should, yes!